js/activity.js

```diff
@@ -2,32 +2,43 @@ define(function (require) {
     var activity = require("sugar-web/activity/activity");
     var icon = require("sugar-web/graphics/icon");
     require("easel");
-    require("handlebars")
-    var shapes = require("activity/shapes")
+    require("handlebars");
+    var shapes = require("activity/shapes");
+
     // Manipulate the DOM only when it is ready.
     require(['domReady!'], function (doc) {
         // Initialize the activity.
         activity.setup();
+
         // Colorize the activity icon.
         var activityButton = document.getElementById("activity-button");
         activity.getXOColor(function (error, colors) {
             icon.colorize(activityButton, colors);
         });
+
+        // Handle New Button Click
         var newButton = document.getElementById("new-button");
         newButton.onclick = function () {
             new_positions();
         }
-        // Make the activity stop with the stop button.
+
+        // Handle Stop Button Click
         var stopButton = document.getElementById("stop-button");
         stopButton.addEventListener('click', function (e) {
             activity.close();
         });
+
+        // Handle Clear Button Click
+        var clearButton = document.getElementById("clear-button"); // **New Line**
+        clearButton.addEventListener('click', clearCanvas); // **New Line**
+
         // Then create a list of the label elements
-        nlabels = [];
-        for (i = 0; i < 21; i++) {
+        var nlabels = [];
```

Comment:
If this is consistent throughout this code, there's no need to change it.

Reply:
sure i will keep them as it is....